<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ProcedureForm</class>
 <widget class="QWidget" name="ProcedureForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>872</width>
    <height>703</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_2">
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QFrame" name="frameForUnlockButton">
     <property name="maximumSize">
      <size>
       <width>59</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="QToolButton" name="btnLock">
        <property name="text">
         <string/>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="popupMode">
         <enum>QToolButton::InstantPopup</enum>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="btnUndelete">
        <property name="toolTip">
         <string>This object was deleted in a Patch. Click this button to Undelete it.</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../dbm.qrc">
          <normaloff>:/images/actions/images/small/gnome_undelete.png</normaloff>:/images/actions/images/small/gnome_undelete.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>48</width>
          <height>48</height>
         </size>
        </property>
        <property name="checkable">
         <bool>false</bool>
        </property>
        <property name="popupMode">
         <enum>QToolButton::InstantPopup</enum>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="grpContent" native="true">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <property name="spacing">
         <number>0</number>
        </property>
        <item>
         <widget class="QToolButton" name="btnLoad">
          <property name="maximumSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Load a .SQL file</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../dbm.qrc">
            <normaloff>:/images/actions/images/actions/fileopen.png</normaloff>:/images/actions/images/actions/fileopen.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="btnSave">
          <property name="maximumSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Save the script to a .SQL file</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../dbm.qrc">
            <normaloff>:/images/actions/images/actions/filesave.png</normaloff>:/images/actions/images/actions/filesave.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="btnInject">
          <property name="maximumSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Inject this procedure in a database</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../dbm.qrc">
            <normaloff>:/images/actions/images/small/database.png</normaloff>:/images/actions/images/small/database.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>16</width>
            <height>16</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QFormLayout" name="formLayout_2">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="lblProcName">
          <property name="text">
           <string>Name</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="txtProcName"/>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGroupBox" name="grpParameters">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>220</height>
         </size>
        </property>
        <property name="title">
         <string> Parameters </string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QComboBox" name="comboBox_2">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The type of this parameter:&lt;/p&gt;&lt;ul style=&quot;margin-top: 0px; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; -qt-list-indent: 1;&quot;&gt;&lt;li style=&quot; margin-top:12px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;IN&lt;/span&gt; This is an incoming parameter. Passed in to the function. &lt;/li&gt;&lt;li style=&quot; margin-top:0px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;OUT&lt;/span&gt; This parameter is an outgoing parameter. Will be used in the caller &lt;/li&gt;&lt;li style=&quot; margin-top:0px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;INOUT&lt;/span&gt; Same parameter used for in and output operations&lt;/li&gt;&lt;/ul&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <item>
              <property name="text">
               <string/>
              </property>
             </item>
             <item>
              <property name="text">
               <string>IN</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>OUT</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>INOUT</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="txtParamName">
             <property name="toolTip">
              <string>The name of the parameter</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="cmbParameterType">
             <property name="toolTip">
              <string>The type of the parameter</string>
             </property>
             <property name="editable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="toolButton">
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="../dbm.qrc">
               <normaloff>:/images/actions/images/small/add.png</normaloff>:/images/actions/images/small/add.png</iconset>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_8">
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <property name="spacing">
              <number>0</number>
             </property>
             <item>
              <widget class="QToolButton" name="toolButton_2">
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset resource="../dbm.qrc">
                 <normaloff>:/images/actions/images/small/up.png</normaloff>:/images/actions/images/small/up.png</iconset>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="toolButton_4">
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset resource="../dbm.qrc">
                 <normaloff>:/images/actions/images/small/down.png</normaloff>:/images/actions/images/small/down.png</iconset>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="toolButton_3">
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset resource="../dbm.qrc">
                 <normaloff>:/images/actions/images/small/remove.png</normaloff>:/images/actions/images/small/remove.png</iconset>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="verticalSpacer">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QListWidget" name="listWidget"/>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_5">
             <property name="spacing">
              <number>0</number>
             </property>
             <item>
              <widget class="QToolButton" name="btnShowDescriptionField">
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset resource="../dbm.qrc">
                 <normaloff>:/images/actions/images/small/pencil3.png</normaloff>:/images/actions/images/small/pencil3.png</iconset>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
               <property name="checked">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="verticalSpacer_3">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QTextEdit" name="txtParamDesc">
             <property name="toolTip">
              <string>Type in a detailed description for the parameter</string>
             </property>
             <property name="documentTitle">
              <string>Description</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4"/>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QFormLayout" name="formLayout_3">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="lblReturnType">
          <property name="text">
           <string>Return type</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="cmbReturnType">
          <property name="editable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="spacing">
         <number>0</number>
        </property>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../dbm.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>btnLock</sender>
   <signal>toggled(bool)</signal>
   <receiver>ProcedureForm</receiver>
   <slot>onLockUnlock(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>38</x>
     <y>37</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnUndelete</sender>
   <signal>clicked()</signal>
   <receiver>ProcedureForm</receiver>
   <slot>onUndelete()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>38</x>
     <y>91</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnLoad</sender>
   <signal>clicked()</signal>
   <receiver>ProcedureForm</receiver>
   <slot>onLoadFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>86</x>
     <y>108</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnSave</sender>
   <signal>clicked()</signal>
   <receiver>ProcedureForm</receiver>
   <slot>onSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>112</x>
     <y>108</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnInject</sender>
   <signal>clicked()</signal>
   <receiver>ProcedureForm</receiver>
   <slot>onInject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>138</x>
     <y>108</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmbReturnType</sender>
   <signal>editTextChanged(QString)</signal>
   <receiver>ProcedureForm</receiver>
   <slot>onReturnTypeComboChange(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>376</x>
     <y>364</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtProcName</sender>
   <signal>textEdited(QString)</signal>
   <receiver>ProcedureForm</receiver>
   <slot>onProcNameChange(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>357</x>
     <y>82</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnShowDescriptionField</sender>
   <signal>toggled(bool)</signal>
   <receiver>txtParamDesc</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>488</x>
     <y>342</y>
    </hint>
    <hint type="destinationlabel">
     <x>680</x>
     <y>393</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onLockUnlock(bool)</slot>
  <slot>onUndelete()</slot>
  <slot>onLoadFile()</slot>
  <slot>onSave()</slot>
  <slot>onNew()</slot>
  <slot>onInject()</slot>
  <slot>onReturnTypeComboChange(QString)</slot>
  <slot>onProcNameChange(QString)</slot>
 </slots>
</ui>
